q2_demux/tests/test_filter.py

```diff
@@ -0,0 +1,183 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2019, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import unittest
+import os
+import pandas as pd
+import q2_demux._filter as filter
+
+from qiime2.plugin.testing import TestPluginBase
+from q2_demux._summarize import _PlotQualView
+from q2_types.per_sample_sequences import (
+    SingleLanePerSampleSingleEndFastqDirFmt,
+    SingleLanePerSamplePairedEndFastqDirFmt)
+
+from qiime2 import Metadata
+
+
+class FilterSamplesTests(TestPluginBase):
+
+    package = 'q2_demux.tests'
+
+    def setUp(self):
+        super().setUp()
+
+        self.data_single = SingleLanePerSampleSingleEndFastqDirFmt(
+            self.get_data_path('filter_samples_single_end/dir_fmt'), mode='r')
+        self.sample_single = _PlotQualView(self.data_single, False)
+        self.manifest_single = self.data_single.manifest.view(pd.DataFrame)
+
+        self.meta_single_all = Metadata.load(
+            self.get_data_path('filter_samples_single_end/filter_all.tsv'))
+        self.meta_single_subset = Metadata.load(
+            self.get_data_path('filter_samples_single_end/filter_subset.tsv'))
+        self.meta_single_none = Metadata.load(
+            self.get_data_path('filter_samples_single_end/filter_none.tsv'))
+
+        self.data_paired = SingleLanePerSamplePairedEndFastqDirFmt(
+            self.get_data_path('filter_samples_paired_end/dir_fmt'), mode='r')
+        self.sample_paired = _PlotQualView(self.data_paired, True)
+        self.manifest_paired = self.data_paired.manifest.view(pd.DataFrame)
+
+        self.meta_paired_all = Metadata.load(
+            self.get_data_path('filter_samples_single_end/filter_all.tsv'))
+        self.meta_paired_subset = Metadata.load(
+            self.get_data_path('filter_samples_single_end/filter_subset.tsv'))
+        self.meta_paired_none = Metadata.load(
```

Comment:
`meta` is a bit confusing, we have a shorthand of `md` which I think will be more clear (borrowing from the convention of DataFrames being called `df`).